recipes-extended/lsb/lsbinitscripts/0001-functions-avoid-exit-1-which-causes-init-scripts-to-.patch

```diff
@@ -0,0 +1,21 @@
+Upstream-Status: Pending
```

Comment:
Has this patch been upstreamed to lsbinitscripts? If so, is there anything stopping us from bumping the source version of the recipe, rather than carrying this `.patch`?

Reply:
It was never upstreamed so we'd still need this patch.

Reply:
Was it never even submitted to the sysvinit upstream?

The more I look at this .patch, the more I don't understand it. I don't see how the `strstr` function would "exit 1" in any case. Do you have an understanding of what Chen Qi was trying to fix here?

Reply:
Don't see any PR (open/closed) in upstream repo, so doesn't look like it was submitted.

[strstr](https://github.com/fedora-sysv/initscripts/blob/master/etc/rc.d/init.d/functions#L602) returns 1 when $1 doesn't contain $2 ([ref](https://stackoverflow.com/a/41806827)). So seems like a legitimate patch.

Reply:
`exit` and `return` are unrelated calls in bash. The `strstr` function returning 0 or 1 shouldn't ever cause either `functions` or a sourcing script to `exit` with code 1, *unless* the returning function is the last call which is made in the script. In which case, the script would `exit` with the last return value.

But since the final call in the script is always `return 0`, I think the bash exit code should always be 0 (unless there is a legitimate bash error).

So, I'm still not convinced this does something. Have you found any evidence of a reproducing case for this change? If it is legitimate, I would like us to adopt the patch and try to upstream it.

Reply:
You're right. I tested it out and the patch isn't necessary. Don't know why it was added in first place.
Will remove it.